node_modules/oae-zencoder/lib/processor.js

```diff
@@ -0,0 +1,130 @@
+/*!
+ * Copyright 2014 Apereo Foundation (AF) Licensed under the
+ * Educational Community License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may
+ * obtain a copy of the License at
+ *
+ *     http://opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+var _ = require('underscore');
+var fs = require('fs');
+var request = require('request');
+var util = require('util');
+
+var ContentUtil = require('oae-content/lib/internal/util');
+var ContentConstants = require('oae-content/lib/constants').ContentConstants;
+var Context = require('oae-context').Context;
+var log = require('oae-logger').logger('oae-zencoder');
+var PreviewConstants = require('oae-preview-processor/lib/constants');
+
+var ZencoderDAO = require('./internal/dao');
+var ZencoderUtil = require('./util');
+
+/**
+ * @borrows Interface.test as Videos.test
+ */
+var test = module.exports.test = function(ctx, content, callback) {
+    if (content.resourceSubType === 'file') {
+        if (_isVideo(ctx.revision.mime) || _isAudio(ctx.revision.mime)) {
+            return callback(null, 30);
+        }
+    }
+
+    return callback(null, -1);
+};
+
+/**
+ * @borrows Interface.generatePreviews as Videos.generatePreviews
+ */
+var generatePreviews = module.exports.generatePreviews = function(ctx, content, callback) {
+    var revision = ctx.revision;
+
+    if (revision.previews && revision.previews.video) {
+        return callback();
+    }
+
+    log().info('Generating previews for video file: %s', content.id);
+    var tenant = content.tenant;
+
+    var downloadPath = ContentUtil.getSignedDownloadUrl(new Context(tenant), revision.uri, 6000, 6000);
+
+    // Create a Zencoder job
+    ZencoderUtil.job(tenant.alias, downloadPath, content.id, ctx.revisionId, _isVideo(revision.mime), function(err, res, body) {
```

Comment:
Rather than providing a boolean `_isVideo` to ZencoderUtil here, can we provide the mime and let ZencoderUtil do it's thing? We'll then need to expose `isVideo` and `isAudio` to the util, so we can probably have `ZencoderUtil.isVideo(mimeType)` as a function.